dev-requirements.txt

```diff
@@ -35,7 +35,9 @@ sqlalchemy-utils~=0.41.2
 scikit-learn~=1.5.1
 lightgbm~=4.3
 xgboost~=1.1
-tensorflow~=2.15
+# TODO: once tensorflow upgrades their dependencies to support a newer version protobuf, we can uncomment this line
+#tensorflow~=2.19; python_version > "3.9"
+tensorflow~=2.15; python_version <= "3.9"
```

Comment:
these changes along with grpcio tools and dask - worth another PR
as this pr mixes both
1. enable dual build of mlrun kfp
2. format dependencies for python 3.9 / python 3.11 needs